packages/toolkit/src/query/tests/queryFn.test.tsx

```diff
@@ -9,9 +9,9 @@ import type { QuerySubState } from '@reduxjs/toolkit/dist/query/core/apiState'
 
 describe('queryFn base implementation tests', () => {
   const baseQuery: BaseQueryFn<string, { wrappedByBaseQuery: string }, string> =
-    jest.fn((arg: string) => ({
-      data: { wrappedByBaseQuery: arg },
-    }))
+    jest.fn((arg: string) => arg.includes('withErrorQuery')
+      ? ({ error: `cut${arg}` })
+      : ({ data: { wrappedByBaseQuery: arg } }))
```

Comment:
something is happening here that i do not understand

I've tried fully mimicking the data case like this
```ts
const baseQuery: BaseQueryFn<string, { wrappedByBaseQuery: string }, { wrappedByBaseQuery: string }> =
    jest.fn((arg: string) => arg.includes('withErrorQuery')
      ? ({ error: { wrappedByBaseQuery: arg } })
      : ({ data: { wrappedByBaseQuery: arg } }))
```

but this causes cases like `withErrorQueryFn`
```ts
withErrorQueryFn: build.query<string, string>({
  queryFn(arg: string) {
    return { error: `resultFrom(${arg})` }
  },
}),
```
 to fail the typecheck
> Type '(arg: string) => { error: string; }' is not assignable to type '(arg: string, api: BaseQueryApi, extraOptions: {}, baseQuery: (arg: string) => MaybePromise<QueryReturnValue<{ wrappedByBaseQuery: string; }, { wrappedByBaseQuery: string; }, {}>>) => MaybePromise<...>'. 
  Type '{ error: string; }' is not assignable to type 'MaybePromise<QueryReturnValue<string, { wrappedByBaseQuery: string; }, unknown>>'. 
    Property 'data' is missing in type '{ error: string; }' but required in type '{ error?: undefined; data: string; meta?: unknown; }'.

i lack proper TS knowledge here so I just went with not changing the `string` error response type. But I'll be glad to do it better if someone hints me on what might be the issue